sklearn/ensemble/_iforest.py

```diff
@@ -522,12 +522,19 @@ def _compute_score_samples(self, X, subsample_features):
         return scores
 
     def _more_tags(self):
+        allow_nan = isinstance(
+            self.estimator, ExtraTreeRegressor
+        ) and self.estimator.splitter in {
+            "best",
+            "random",
+        }
```

Comment:
I don't think that we allow any other type of trees than `ExtraTreeRegressor` because this is not a parameter that is supposed to be set. So I would always set it to `True`.

Reply:
Oh right good catch!